stats/src/test/java/com/facebook/stats/mx/TestJVMStatsExporter.java

```diff
@@ -62,6 +64,30 @@ public void testFilteredStats() throws Exception {
     Assert.assertTrue(exportedStats.containsKey("jvm.MemoryPool.Code_Cache.UsageThresholdCount"));
     Assert.assertTrue(exportedStats.containsKey("jvm.MemoryPool.Code_Cache.PeakUsage.committed"));
   }
+
+  @Test(groups = "fast")
+  public void testStatNameReplacer() throws Exception {
+    Stats stats = new Stats();
+    // Chose an MBean that has a good chance of being there across different jvm versions
+    JVMStatsExporter jvmStatsExporter = new JVMStatsExporter(
+      stats,
+      (bean, attribute, keys) -> {
+        StringBuilder name = new StringBuilder();
+        name.append("test");
+        name.append(".").append(attribute);
+        for (String attributeName : keys) {
+          name.append('.').append(attributeName);
+        }
+
+        return Optional.of(name.toString());
+      },
+      "java.lang:type=MemoryPool,name=Code Cache"
+    );
+    Map<String, Long> exportedStats = getExportedStats(stats);
+    for (Map.Entry<String, Long> entry : exportedStats.entrySet()) {
```

Comment:
Instead of iterating over all the stats, better to assert a specific stat name, e.g.

    Assert.assertNotNull(exportedStats.get("test.Usage.used"), "test.Usage.used");
    Assert.assertNotNull(exportedStats.get("test.Usage.used"), "test.PeakUsage.max");

The current test would pass if no stats were exported.